docs/concepts/crds/backupconfiguration.md

```diff
@@ -0,0 +1,201 @@
+---
+title: BackupConfiguration Overview
+menu:
+  product_stash_0.8.3:
+    identifier: backupconfiguration-overview
+    name: BackupConfiguration
+    parent: crds
+    weight: 15
+product_name: stash
+menu_name: product_stash_0.8.3
+section_menu_id: concepts
+---
+
+> New to Stash? Please start [here](/docs/concepts/README.md).
+
+# BackupConfiguration
+
+## What is BackupConfiguration
+
+A `BackupConfiguration` is a Kubernetes `CustomResourceDefinition (CRD)` which specifies the backup target, behaviors (schedule, retention policy etc.) and `Repository` object that holds backend information in Kubernetes native way.
+
+You have to create a `BackupConfiguration` object for each backup target. `BackupConfiguration` has 1-1 mapping with the target. Thus, only one target can be backed up using one `BackupConfiguration`.
+
+## BackupConfiguration CRD Specification
+
+Like any official Kubernetes resource, a `BackupConfiguration` has `TypeMeta`, `ObjectMeta` and `Spec` sections. However, unlike other Kubernetes resources, it does not have a `Status` section.
+
+A sample `BackupConfiguration` object to backup a Deployment's data is shown below,
+
+```yaml
+apiVersion: stash.appscode.com/v1beta1
+kind: BackupConfiguration
+metadata:
+  name: demo-backup
+  namespace: demo
+spec:
+  driver: Restic
+  repository:
+    name: local-repo
+  # task:
+  #   name: workload-backup # task field is not required for workload data backup but it is necessary for database backup.
+  schedule: "* * * * *" # backup at every minutes
+  paused: false
+  target:
+    ref:
+      apiVersion: apps/v1
+      kind: Deployment
+      name: stash-demo
+    directories:
+    - /source/data
+    volumeMounts:
+    - name: source-data
+      mountPath: /source/data
+  runtimeSettings:
+    container:
+      resources:
+        requests:
+          memory: 256M
+        limits:
+          memory: 256M
+      securityContext:
+        runAsUser: 2000
+        runAsGroup: 2000
+      nice:
+        adjustment: 5
+      ionice:
+        class: 2
+        classData: 4
+    pod:
+      imagePullSecrets:
+      - name:  my-private-registry-secret
+      serviceAccountName: my-backup-svc
+  tempDir:
+    medium: "Memory"
+    size:  "2Gi"
+    disableCache: false
+  retentionPolicy:
+    name: 'keep-last-5'
+    keepLast: 5
+    prune: true
+```
+
+Here, we are going to describe some important sections of `BackupConfiguration` crd.
+
+### BackupConfiguration `Spec` Section
+
+`BackupConfiguration` object holds following fields in `.spec` section.
+
+#### spec.driver
+
+`spec.driver` indicates the tool to use to backup the target. Currently, Stash accept `Restic` and `VolumeSnapshotter` as driver. The default value of this field is `Restic`.
```

Comment:
Do we use mutator to set this field to Restic if not set?

Reply:
No. We don't. If this field is kept empty, we assume it is `Restic`. You have asked not to  add mutating webhook for CRDs for now.

Reply:
Filed https://github.com/stashed/stash/issues/779